pkgs/applications/misc/llpp/default.nix

```diff
@@ -1,48 +1,51 @@
-{ stdenv, lib, makeWrapper, fetchgit, pkgconfig, ninja, ocaml, findlib, mupdf
-, gtk3, openjpeg, jbig2dec, mujs, xsel, openssl, freetype, ncurses }:
+{ stdenv, lib, makeWrapper, fetchgit, ocaml, mupdf, libX11, libGLU_combined,
+freetype, xclip }:
 
-assert lib.versionAtLeast (lib.getVersion ocaml) "4.02";
+assert lib.versionAtLeast (lib.getVersion ocaml) "4.07";
 
 stdenv.mkDerivation rec {
   name = "llpp-${version}";
-  version = "2018-03-02";
+  version = "30";
 
   src = fetchgit {
     url = "git://repo.or.cz/llpp.git";
-    rev = "0ab1fbbf142b6df6d6bae782e3af2ec50f32dec9";
-    sha256 = "1h0hrmxwm7ripgp051788p8ad0q38dc9nvjx87mdwlkwk9qc0dis";
+    rev = "v${version}";
+    sha256 = "0iilpzf12hs0zky58j55l4y5dvzv7fc53nsrg324n9vka92mppvd";
     fetchSubmodules = false;
   };
 
-  nativeBuildInputs = [ pkgconfig makeWrapper ninja ];
-  buildInputs = [ ocaml findlib mupdf gtk3 jbig2dec openjpeg mujs openssl freetype ncurses ];
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ ocaml mupdf libX11 libGLU_combined freetype ];
 
   dontStrip = true;
 
   configurePhase = ''
-    sed -i -e 's+ocamlc --version+ocamlc -version+' build.sh
-    sed -i -e 's+-I \$srcdir/mupdf/include -I \$srcdir/mupdf/thirdparty/freetype/include+-I ${freetype.dev}/include+' build.sh
-    sed -i -e 's+-lmupdf +-lfreetype -lz -lharfbuzz -ljbig2dec -lopenjp2 -ljpeg -lmupdf +' build.sh
-    sed -i -e 's+-L\$srcdir/mupdf/build/native ++' build.sh
+    sed -i -e '/test -d "$mudir" || die muPDF not found, consult $(dirname $0)\/BUILDING/d' build.bash
+    sed -i -e 's+-I $mudir/include -I $mudir/thirdparty/freetype/include+-I ${freetype.dev}/include+' build.bash
+    sed -i -e '/isfresh "$mulibs"/,+3d' build.bash
+    # TODO: remove after v31 comes out
+    sed -i -e 's+4.07.0+4.07.1+' build.bash
+    sed -i -e '/ver=/c\ver=v${version}' build.bash
+    sed -i -e 's+-L$mudir/build/$mbt -lmupdf -lmupdf-third +-lmupdf -lfreetype +' build.bash
```

Comment:
Can you make this a patch?

This is an example to interpolate dynamic variables:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/click/default.nix#L12

Apart from that, you could try to avoid the freetype patch by symlinking its directory to `mudir/thirdparty/freetype` and adding `-lfreetype` to the `NIX_CFLAGS_COMPILE` environment variable.

Reply:
Updated 👍